opba-auth-rest-impl/src/main/java/de/adorsys/opba/tppauthapi/service/PsuAuthService.java

```diff
@@ -0,0 +1,36 @@
+package de.adorsys.opba.tppauthapi.service;
+
+import de.adorsys.datasafe.encrypiton.api.types.UserIDAuth;
+import de.adorsys.opba.db.domain.entity.psu.Psu;
+import de.adorsys.opba.db.repository.jpa.psu.PsuRepository;
+import de.adorsys.opba.protocol.facade.config.encryption.impl.psu.PsuSecureStorage;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+
+import javax.transaction.Transactional;
+import java.util.Optional;
+
+@Service
+@RequiredArgsConstructor
+public class PsuAuthService {
+
+    private final PsuRepository psuRepository;
+    private final PsuSecureStorage psuSecureStorage;
+
+
+    public Optional<Psu> getPsu(String psuId) {
+        return psuRepository.findByUserId(psuId);
+    }
+
+    @Transactional
+    public Optional<Psu> createPsuIfNotExist(String id, String password) {
```

Comment:
As creating already existing user seem to be exceptional case, suggest to throw an exception that is handled by ControllerAdvice instead of returning optional.

Reply:
done